### PR TITLE
Simple e2e testing with kind

### PR DESCRIPTION
**Pull request by dimaunx**
1. Add support for e2e tests with kind. make e2e command
2. Add support for running go test.

image

### CODE REVIEW

**Comment by dfarrell07** [comment]
I tested this on my system and it worked perfectly out of the box. :+1:

File: Dockerfile.dapper [hunk 1]
```diff
@@ -3,22 +3,31 @@ FROM ubuntu:16.04
...
     GOPATH=/go PATH=/go/bin:/usr/local/go/bin:${PATH} SHELL=/bin/bash
 
-RUN wget -O - https://storage.googleapis.com/golang/go1.11.linux-${!GOLANG_ARCH}.tar.gz | tar -xzf - -C /usr/local && \
+RUN wget -q -O - https://storage.googleapis.com/golang/go1.11.linux-${!GOLANG_ARCH}.tar.gz | tar -xzf - -C /usr/local && \
     go get github.com/rancher/trash
 
-ENV DOCKER_URL_amd64=https://get.docker.com/builds/Linux/x86_64/docker-1.10.3 \
-    DOCKER_URL_arm=https://github.com/rancher/docker/releases/download/v1.10.3-ros1/docker-1.10.3_arm \
-    DOCKER_URL=DOCKER_URL_${ARCH}
+RUN curl -LO https://storage.googleapis.com/kubernetes-release/release/$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/linux/amd64/kubectl && \
```
Comment (skitt) [inline Dockerfile.dapper]: Wouldn’t it be better to use `wget` or `curl` throughout, not both?

File: scripts/lint [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env bash
+set -e
+
+cd $(dirname $0)/..
+
+golangci-lint run -D megacheck -E unused,gosimple,staticcheck
```
Comment (skitt) [inline scripts/lint]: I think this should be an edit to `scripts/validate` (which currently lints using `golint`), but as a separate PR.

File: Dockerfile.dapper (hunk 1 above)
Comment (dimaunx) [reply Dockerfile.dapper]: Good suggestion, curl now is used in all commands and RUN block was consolidated to keep intermediate less layers.

File: scripts/lint (hunk 2 above)
Comment (dimaunx) [reply scripts/lint]: Removed the lint script. Agree, we should do it in separate PR.

File: scripts/kind-e2e/cluster2-config.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  metadata:
+    name: config
+  networking:
+    podSubnet: 10.245.0.0/16
+    serviceSubnet: 100.95.0.0/16
+    dnsDomain: cluster2.local
+nodes:
+- role: control-plane
+- role: worker
+- role: worker
```
Comment (tpantelis) [inline scripts/kind-e2e/cluster2-config.yaml]: duplicate of line 15 - is this intended? ...

File: scripts/kind-e2e/cluster3-config.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  metadata:
+    name: config
+  networking:
+    podSubnet: 10.246.0.0/16
+    serviceSubnet: 100.96.0.0/16
+    dnsDomain: cluster3.local
+nodes:
+- role: control-plane
+- role: worker
+- role: worker
```
Comment (tpantelis) [inline scripts/kind-e2e/cluster3-config.yaml]: ... same here

File: scripts/kind-e2e/cluster2-config.yaml (hunk 3 above)
Comment (skitt) [reply scripts/kind-e2e/cluster2-config.yaml]: Yes, we want two worker nodes.

File: scripts/kind-e2e/cluster3-config.yaml (hunk 4 above)
Comment (dimaunx) [reply scripts/kind-e2e/cluster3-config.yaml]: Yes this this is kind configuration for 2 worker nodes.

**Comment by dfarrell07** [comment]
I tested this again locally at 70fa657 and it still worked.

Imho these should be merged into one commit. I'd also love that commit to have a more detailed commit message, following the [pattern well-described here](https://chris.beams.io/posts/git-commit/).

Very good work! :+1:

**Pull request merged**